examples/loader-rust-to-json/src/index.md

```diff
@@ -0,0 +1,232 @@
+        let is_straight = if self.0.iter().any(|card| card.rank == 1) {
+            // Handle aces
+            let min_rank = self.0.iter().map(|card| card.rank).filter(|&rank| rank != 1).min().unwrap();
+            let max_rank = self.0.iter().map(|card| card.rank).filter(|&rank| rank != 1).max().unwrap();
+            (min_rank == 2 && max_rank == 5) || (min_rank == 10 && max_rank == 13)
+        } else {
+            let min_rank = self.0.iter().map(|card| card.rank).min().unwrap();
+            let max_rank = self.0.iter().map(|card| card.rank).max().unwrap();
+            (max_rank - min_rank) as usize == self.0.len() - 1
+        };
+
+        if is_flush && is_straight {
+            HandCategory::StraightFlush
+        } else if rank_counts.values().any(|&count| count == 4) {
+            HandCategory::FourOfAKind
+        } else if rank_counts.values().any(|&count| count == 3)
+            && rank_counts.values().any(|&count| count == 2)
+        {
+            HandCategory::FullHouse
+        } else if is_flush {
+            HandCategory::Flush
+        } else if is_straight {
+            HandCategory::Straight
+        } else if rank_counts.values().any(|&count| count == 3) {
+            HandCategory::ThreeOfAKind
+        } else if rank_counts.values().filter(|&&count| count == 2).count() == 2 {
+            HandCategory::TwoPair
+        } else if rank_counts.values().any(|&count| count == 2) {
+            HandCategory::OnePair
+        } else {
+            HandCategory::HighCard
+        }
+    }
+}
+```
+
+<div class="note">
+
+To run this data loader, you’ll need [rust-script](https://rust-script.org/) installed.
+
+</div>
+
+The above data loader lives in `data/poker.json.rs`, so we can load the data as `data/poker.json` using `FileAttachment`.
+
+```js echo
+const hands = FileAttachment("data/poker.json").json();
+```
+
+<div style="max-width: 300px;">
+
+```js
+Inputs.table(hands.summary)
+```
+
+</div>
+
+Taking advantage of Rust’s performance and simple parallelism, the data loader above was able to summarize ${hands.meta.count / 1e6} million hands in ${hands.meta.duration_s.toFixed(1)} seconds.
+
+We can make a quick chart of the poker hands with Plot
+
+```js echo
+Plot.plot({
+  x: {tickFormat: "%", grid: true},
+  y: {domain: hands.summary.map(d => d.category)},
+  marginLeft: 100,
+  marginRight: 30,
+  marks: [
+    Plot.ruleX([0]),
+    Plot.barX(hands.summary, {
+      x: d => d.count / hands.meta.count,
+      y: "category",
+      fill: "category",
+    }),
+    Plot.textX(hands.summary, {
+      x: d => d.count / hands.meta.count,
+      y: "category",
+      text: d => formatPercent(d.count / hands.meta.count),
+      textAnchor: "start",
+      dx: 2,
+    }),
+  ]
+})
+```
+
+```js echo
+function formatPercent(x) {
+  // Displays both large and small percentages by adding decimal places
+  // until there is something besides zeroes, or until the limit is hit.
```

Comment:
Neat! Another “trick” I like with percentages is to not round 0.99999 to 100% but display it as `99.9%` (not applicable here, but I can see this function being reused in other contexts).